prov/gni/src/gnix_cm_nic.c

```diff
@@ -190,24 +190,28 @@ static int __process_datagram(struct gnix_datagram *dgram,
 /*******************************************************************************
  * Internal API functions
  ******************************************************************************/
-/*
- * generate a cdm_id, use the 16 LSB of base_id from domain
- * with 16 MSBs being obtained from atomic increment of
- * a local variable.
- */
 
-int _gnix_get_new_cdm_id(struct gnix_fid_domain *domain, uint32_t *id)
+int _gnix_cm_nic_create_cdm_id(struct gnix_fid_domain *domain, uint32_t seed,
+			       uint32_t *id)
 {
 	uint32_t cdm_id;
-	int v;
 
-	v = atomic_inc(&gnix_id_counter);
 	cdm_id = (domain->cdm_id_seed & 0x0000FFFF) |
```

Comment:
Is the format of cdm_id documented somewhere?  How come the first two bytes of cdm_id_seed are masked off?